authors/apps/authentication/views.py

```diff
@@ -230,3 +237,98 @@ def list(self, request):
+        serializer.is_valid(raise_exception=True)
+
+        provider = serializer.data.get('provider', None)
+        strategy = load_strategy(request)  # creates the app instance
+
+        if request.user.is_anonymous:  # make sure the user is not anonymous
+            user = None
+        else:
+            user = request.user
+
+        try:
+            # load backend with strategy and provider from settings(AUTHENTICATION_BACKENDS)
+            backend = load_backend(
+                strategy=strategy, name=provider, redirect_uri=None)
+
+        except MissingBackend as error:
+
+            return Response({
+                "errors": str(error)
+            }, status=status.HTTP_400_BAD_REQUEST)
+
+        try:
+            # check type of oauth provider e.g facebook is BaseOAuth2 twitter is BaseOAuth1
+            if isinstance(backend, BaseOAuth1):
+                # oath1 passes access token and secret
+                access_token = {
+                    "oauth_token": serializer.data.get('access_token'),
+                    "oauth_token_secret": serializer.data.get('access_token_secret'),
+                }
+
+            elif isinstance(backend, BaseOAuth2):
+                # oauth2 only has access token
+                access_token = serializer.data.get('access_token')
+
+            user = backend.do_auth(access_token)
+
+        except HTTPError as error:
+            return Response({
+                "error": {
+                    "access_token": "invalid token",
+                    "details": str(error)
+                }
+            }, status=status.HTTP_400_BAD_REQUEST)
+
+        except AuthTokenError as error:
+            return Response({
+                "error": "invalid credentials",
+                "details": str(error)
+            }, status=status.HTTP_400_BAD_REQUEST)
+
+        try:
+            # authenticate the current user
+            # social pipeline associate by email handles already associated exception
+            if isinstance(backend, BaseOAuth2):
+                # oauth2 only has access token
+                access_token = serializer.data.get('access_token')
+                user = backend.do_auth(access_token)
+
+            authenticated_user = backend.do_auth(access_token, user=user)
+
+        except HTTPError as error:
+            # catch any error as a result of the authentication
+            return Response({
+                "error": "Http Error",
+                "details": str(error)
+            }, status=status.HTTP_400_BAD_REQUEST)
+
+        except AuthForbidden as error:
+            return Response({
+                "error": "invalid token",
+                "details": str(error)
+            }, status=status.HTTP_400_BAD_REQUEST)
+
+        if authenticated_user and authenticated_user.is_active:
+            # Check if the user you intend to authenticate is active
+            token = JWTokens.create_token(self, authenticated_user)
```

Comment:
Why not use the `token` property in the user model so that we can have `authenticated_user.token`

Reply:
Thanks for the feedback It's done